semantic/monotype.go

```diff
@@ -263,25 +268,53 @@ func (mt MonoType) ReturnType() (MonoType, error) {
 	return NewMonoType(tbl, f.RetnType())
 }
 
-func getArr(tbl fbTabler) (*fbsemantic.Arr, error) {
-	arr, ok := tbl.(*fbsemantic.Arr)
+// ElemType returns the element type if this monotype is an array or vector, and an error otherise
+func (mt MonoType) ElemType() (MonoType, error) {
+	c, ok := getCollection(mt.tbl)
 	if !ok {
-		return nil, errors.New(codes.Internal, "MonoType is not an array")
+		return MonoType{}, errors.New(codes.Internal, "MonoType is not a collection")
 	}
-	return arr, nil
+	return c.ElementType()
 }
 
-// ElemType returns the element type if this monotype is an array, and an error otherise.
-func (mt MonoType) ElemType() (MonoType, error) {
-	arr, err := getArr(mt.tbl)
-	if err != nil {
-		return MonoType{}, err
-	}
+type Collection interface {
```

Comment:
Can you make this an unexported interface? Just make it lowercase.

Reply:
Can do.